### PR TITLE
feat: lv9x migration

### PR DESCRIPTION
**Pull request by romanetar**
ref https://tipit.avaza.com/project/view#!tab=task-pane&groupby=MyTaskDueDate&view=vertical&task=3321351&stafilter=NotComplete&fileview=grid

Signed-off-by: romanetar [roman_ag@hotmail.com]

### CODE REVIEW

File: app/Http/Middleware/TrustProxies.php [hunk 1]
```diff
@@ -19,5 +19,10 @@ class TrustProxies extends Middleware
      *
      * @var int
      */
-    protected $headers = Request::HEADER_X_FORWARDED_ALL;
+    protected $headers =
```
Comment (smarcet) [inline app/Http/Middleware/TrustProxies.php:22]: what is the rationale of this change ?

File: composer.json [hunk 2]
```diff
@@ -29,44 +23,44 @@
...
         "sokil/php-isocodes": "^3.0",
-        "spatie/flysystem-dropbox": "^1.2",
+        "spatie/flysystem-dropbox": "^3.0.1",
         "stripe/stripe-php": "^7.95.0",
         "symfony/yaml": "^5.0",
         "tecnickcom/tcpdf": "^6.2",
-        "vladimir-yuldashev/laravel-queue-rabbitmq": "v11.3.*",
-        "zendframework/zend-math": "^3.2"
+        "vladimir-yuldashev/laravel-queue-rabbitmq": "13.3.5",
+        "laminas/laminas-math": "^3.7"
```
Comment (smarcet) [inline composer.json:54]: please explain why "zendframework/zend-math" its needed for crypto and its a dep of other libs
this will break functionality

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    nano \
+    python3 \
+    make \
+    g++\
+    gpg \
+    gettext
+
+RUN apt clean && rm -rf /var/lib/apt/lists/*
+
+RUN docker-php-ext-install pdo_mysql mbstring exif pcntl bcmath sockets gettext
```
Comment (smarcet) [inline Dockerfile:31]: please check missing exts
 "ext-gd": "*",
        "ext-imagick": "*",
        "ext-json": "*",
        "ext-openssl": "*",
        "ext-pdo": "*",
        "ext-zlib": "*",
from composer.json

File: package.xml [hunk 4]
```diff
@@ -0,0 +1,3908 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (smarcet) [inline package.xml:1]: what is this file for ?
sounds like it should be removed and added it to gitigonre

File: start_local_server.sh [hunk 5]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+set -e
+export DOCKER_SCAN_SUGGEST=false
+
+docker compose run --rm app composer install --ignore-platform-req=ext-gd --ignore-platform-req=ext-imagick
```
Comment (smarcet) [inline start_local_server.sh:5]: why do u need to ignore the listed ext-*  u need to install it on docker please check
https://github.com/OpenStackweb/summit-api/pull/265/files#r1631148173

**Review by smarcet — CHANGES_REQUESTED** [review]
@romanetar please review comments

File: app/Http/Middleware/TrustProxies.php (hunk 1 above)
Comment (romanetar) [reply app/Http/Middleware/TrustProxies.php:22]: Please refer https://laravel.com/docs/9.x/upgrade section Trusted Proxies

File: app/Services/FileSystem/Dropbox/DropboxServiceProvider.php [hunk 6]
```diff
@@ -37,13 +40,18 @@ public function register()
...
      * @return void
      */
-    public function boot()
+    public function boot(): void
     {
         Storage::extend('dropbox', function ($app, $config) {
-            $client = new DropboxClient(
-                $config['authorization_token'] ?? ''
+            $adapter = new DropboxAdapter(
+                new DropboxClient($config['authorization_token'])
```
[the remark this answers is not in the document]
Comment (romanetar) [reply app/Services/FileSystem/Dropbox/DropboxServiceProvider.php:47]: done

File: composer.json (hunk 2 above)
Comment (romanetar) [reply composer.json:54]: Done. All references to Zend were changed to Laminas.

File: Dockerfile (hunk 3 above)
Comment (romanetar) [reply Dockerfile]: All the missing exts were added except for:

- ext-json: no longer needed in this version of PHP

- ext-openssl: adding this extension breaks the build due to this error: "Cannot find config.m4. Make sure that you run '/usr/local/bin/phpize' in the top-level source directory of the module". 
We´re gonna need to dig deeper here.

Comment (smarcet) [reply Dockerfile]: i am gonna re open this
bc its not a feasible solution
we need to know why ext openssl is failing

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+ENV GITHUB_OAUTH_TOKEN=$GITHUB_OAUTH_TOKEN
+ENV PHP_DIR /usr/local/etc/php
+
+# base packages
+RUN apt-get update && apt-get install -y \
+    git \
+    curl \
+    libpng-dev \
+    libonig-dev \
+    libxml2-dev \
```
Comment (smarcet) [inline Dockerfile:18]: lets try to install libssl-dev to fix the ext openssl error

File: Dockerfile (hunk 3 above)
Comment (romanetar) [reply Dockerfile]: This is done

File: composer.json [hunk 8]
```diff
@@ -29,44 +23,44 @@
         "doctrine/annotations": "^1.13",
         "eluceo/ical": "^0.15.0",
         "ezyang/htmlpurifier": "4.13.*",
-        "fideloper/proxy": "^4.4",
         "firebase/php-jwt": "^6.8",
         "fruitcake/laravel-cors": "^2.0",
```
Comment (smarcet) [inline composer.json:27]: @romanetar please review this
Package fruitcake/laravel-cors is abandoned, you should avoid using it. No replacement was suggested.

**Review by smarcet — APPROVED** [review]
LGTM
